src/mlpack/methods/ann/loss_functions/l1_loss.hpp

```diff
@@ -0,0 +1,94 @@
+/**
+ * @file l1_loss.hpp
+ * @author Himanshu Pathak
+ *
+ * Definition of the L1 Loss function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LOSS_FUNCTION_L1_LOSS_HPP
+#define MLPACK_METHODS_ANN_LOSS_FUNCTION_L1_LOSS_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The L1 loss is a loss function that measures the mean absolute error (MAE) 
+ * between each element in the input x and target y 
+ *
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat
+>
+class L1Loss
+{
+ public:
+  /**
+   * Create the L1Loss object.
+   *
+   * @param mean Reduction type. If true, performs mean of loss else sum.
```

Comment:
The last sentence seems grammatically odd. Maybe it should be `If true, it returns the mean of the loss. Else, it returns the sum.`